src/sisl/io/siesta/vibra.py

```diff
@@ -0,0 +1,259 @@
+
+        Parameters
+        -----------
+        close: bool, optional
+            Whether the file unit should be closed afterwards.
+        """
+        self._open()
+        # Read the sizes relevant to the file.
+        # We also read whether there's only gamma point information or there are multiple points
+        if self._state == -1:
+            self._sizes = self._r_next_sizes()
+            self._state = 0
+
+        if close:
+            self._close()
+
+    def _r_next_sizes(self):
+        """Determine the dimension of the data stored in the vectors file"""
+        pos = self.fh.tell()
+        # Skip past header
+        self.readline()  # Empty line
+        self.readline()  # K point
+        self.readline()  # Eigenmode index
+        self.readline()  # Frequency
+        self.readline()  # Eigenmode header
+
+        # Determine number of atoms (i.e. rows per mode)
+        natoms = 0
+        while True:
+            line = self.readline()
+            if line.startswith("Eigenmode"):
+                break
+            else:
+                natoms += 1
+        self._natoms = natoms
+
+        # Determine number of modes
+        nmodes = 1
+        while True:
+            line = self.readline()
+            if line.startswith("k            ="):
+                break
+            elif line.startswith("Eigenvector"):
+                nmodes += 1
+        self._nmodes = nmodes
+
+        # Rewind
+        self.seek(pos)
+
+    def _r_next_eigenmode(self):
+        """Reads the next phonon eigenmode in the vectors file.
+
+        Returns
+        --------
+        EigenmodePhonon:
+            The next eigenmode.
+        """
+        # Skip empty line at the head of the file
+        self.readline()
+
+        k = list(map(float, self.readline().split()[2:]))
+        if len(k) == 0:
+            raise GeneratorExit
+        # Read first eigenvector index
+
+        # Determine number of atoms (i.e. rows per mode)
+        state = np.empty((self._nmodes, self._natoms, 3), dtype=np.complex128)
+        c = np.empty(self._nmodes, dtype=np.float64)
+        for imode in range(self._nmodes):
+            self.readline()
+
+            # Read frequency
+            c[imode] = float(self.readline().split("=")[1])
+
+            # Read real part of eigenmode
+            # Skip eigenmode header
+            self.readline()
+            for iatom in range(self._natoms):
+                line = self.readline()
+                state[imode, iatom, :].real = list(map(float, line.split()))
+
+            # Read imaginary part of eigenmode
+            # Skip eigenmode header
+            self.readline()
+            for iatom in range(self._natoms):
+                line = self.readline()
+                state[imode, iatom, :].imag = list(map(float, line.split()))
+
+        info = dict(k=self._convert_k(k), parent=self.parent)
+        return EigenmodePhonon(state, c * _hc_eV, **info)
```

Comment:
the `c` unit from the file is cm**-1, will this bring it to `eV`? If so, could we name the variable `_cm1_eV` to make it a bit clearer?